### PR TITLE
feat(builder-shared): supports opening specified browser through environment variable

### PR DESCRIPTION
**Pull request by diaz-zeng**
### <samp>🤖 Generated by Copilot at ab949e1</samp>

This pull request adds a new feature to the `@modern-js/builder-shared` package that allows users to open a specified browser through an environment variable `BROWSER` when running the `modern.js` framework. It also updates the documentation and tests for this feature in both English and Chinese.

### <samp>🤖 Generated by Copilot at ab949e1</samp>

*  Add the ability to open a specified browser through an environment variable `BROWSER` for the `@modern-js/builder-shared` package ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-b4a4bf52f5a20f893c30f99300f17fad0623f56e124aaad913a7db39f89fb2beR1-R5))
  * Implement a new function `getTargetBrowser` that returns the browser name based on the environment variable or the running Chromium browsers ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-c3631e7f4f813bd27fa589945623973dfb296addffeb271077b301524cab67e8R25-R35))
  * Use `getTargetBrowser` in the existing function `openBrowser` that tries to open a browser with a given URL ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-c3631e7f4f813bd27fa589945623973dfb296addffeb271077b301524cab67e8L34-R51))
  * Fix a missing `return false` statement in `openBrowser` ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-c3631e7f4f813bd27fa589945623973dfb296addffeb271077b301524cab67e8R59))
  * Add tests for `openBrowser` with different values of `BROWSER` using `vitest` ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-0aeec1d9bd74a444ed9cce3286e224cf8aa1946a16e44c962256ea958650bfbdR1-R13))
* Update the documentation for the `modern.js` framework to explain how to use the `BROWSER` environment variable on MacOS ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-34810316034103fee611a3f98bb257346b3ee6329a5af2e62a1258701e46da1aR34-R46), [link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-7fbef73cd0d3aa4de04c0c5a4177f5c848423494e5a450c1245745fd3833543cR34-R48))
  * Add a new section `Open the specified browser` to the English file `startUrl.md` in the `builder-doc` package ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-34810316034103fee611a3f98bb257346b3ee6329a5af2e62a1258701e46da1aR34-R46))
  * Add a translated section `打开指定浏览器` to the Chinese file `startUrl.md` in the same package ([link](https://github.com/web-infra-dev/modern.js/pull/4358/files?diff=unified&w=0#diff-7fbef73cd0d3aa4de04c0c5a4177f5c848423494e5a450c1245745fd3833543cR34-R48))

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/builder-doc/docs/zh/config/dev/startUrl.md [hunk 1]
```diff
@@ -31,3 +31,18 @@ export default {
   },
 };
 ```
+
+### 打开指定浏览器
+
+在 MacOS 上，通过设置环境变量`BROWSER`，你可以指定 Dev Server 在启动时打开的浏览器，支持如下的值：
```
Comment (chenjiahan) [inline packages/document/builder-doc/docs/zh/config/dev/startUrl.md:37]:
```suggestion
在 MacOS 上，通过设置环境变量 `BROWSER`，你可以指定 Dev Server 在启动时打开的浏览器，支持如下的值：
```

**Review by chenjiahan — DISMISSED** [review]

**Comment by chenjiahan** [comment]
👍🏻 LGTM

**Comment by chenjiahan** [comment]
The test case is failed in Windows, please check it

**Comment by diaz-zeng** [comment]
> The test case is failed in Windows, please check it

Fixed

**Pull request merged**